src/EditorFeatures/VisualBasicTest/Formatting/Indentation/SmartTokenFormatter_FormatTokenTests.vb

```diff
@@ -196,9 +196,10 @@ End Class
                 Dim ignoreMissingToken = previousToken.IsMissing AndAlso line.Start.Position = position
```

Comment:
this local is so weird.  @heejaechang do you know why it's written this way.  this value is *only* computed in tests and then passed down into ShouldUseSmartTokenFormatterInsteadOfIndenter (in the produce this always defaults to true).  Which means that tests are not exercising codepaths that the actual product exercises.  I am *not* a fan of tests calling into implementation details of these types, and then having to shuttle along sideband data to change behavior.

Reply:
this?
http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic.EditorFeatures/Formatting/Indentation/VisualBasicIndentationService.vb,68

can't remember so long since this is written. but probably some corner case we found where we can't actually format due to parse tree shape by vb parser or something.

Reply:
My primary complaint here is that the test is diving into a internal detail of the feature code.  So rather than testing at the top level, it's calling directly into code and depending on hte specific values that are being passed along.